### PR TITLE
[libgpiod] Arm support cxx bindings

### PR DESCRIPTION
**Pull request by Woefie**
This gives support for arm triplets. I don't know why arm64 was chosen but arm works fine.
Also added cxx bindings as a feature

- #### What does your PR fix?
  Fixes arm support. This library is not dependend on arm64 machines. Tested this on am5729 device

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  only linux, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
 Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/libgpiod/portfile.cmake [hunk 1]
```diff
@@ -18,20 +23,25 @@ include("${cmake_vars_file}")
...
 if (VCPKG_DETECTED_CMAKE_CROSSCOMPILING STREQUAL "TRUE")
     list(APPEND OPTIONS CC=${VCPKG_DETECTED_CMAKE_C_COMPILER})
-    if (VCPKG_TARGET_IS_LINUX AND VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
+    if (VCPKG_TARGET_IS_LINUX AND VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
         list(APPEND OPTIONS ac_cv_func_malloc_0_nonnull=yes)
         list(APPEND OPTIONS ac_cv_func_realloc_0_nonnull=yes)
     endif()
 endif()
 
+if(USE_CXX_BINDINGS)
```
Comment (JackBoosY) [inline ports/libgpiod/portfile.cmake:32]:
```suggestion
set(cxx-binding no)
if(USE_CXX_BINDINGS)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libgpiod/portfile.cmake [hunk 2]
```diff
@@ -5,6 +5,11 @@ vcpkg_from_git(
     PATCHES
 )
 
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        cxx-bindings USE_CXX_BINDINGS
+)
```
Comment (JackBoosY) [inline ports/libgpiod/portfile.cmake:11]: I noticed that `FEATURE_OPTIONS` isn't used in anywhere.
Does this only judge that whether this option is enabled?

**Review by JackBoosY — DISMISSED** [review]

**Comment by JackBoosY** [comment]
I can't test this feature since my Ubuntu vm machine can't access the repo. But it LGTM.

File: ports/libgpiod/portfile.cmake [hunk 3]
```diff
@@ -18,25 +18,32 @@ include("${cmake_vars_file}")
 
 if (VCPKG_DETECTED_CMAKE_CROSSCOMPILING STREQUAL "TRUE")
     list(APPEND OPTIONS CC=${VCPKG_DETECTED_CMAKE_C_COMPILER})
-    if (VCPKG_TARGET_IS_LINUX AND VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
+    if (VCPKG_TARGET_IS_LINUX AND VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
```
Comment (ras0219-msft) [inline ports/libgpiod/portfile.cmake:21]: Why is arm64 being replaced by arm here instead of adding additional logic? Does it not work for arm64?

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**